### PR TITLE
refactor(component): exclude empty categories from page

### LINKED ISSUE #385: Exclude category or category group if empty
### Subject

Category

### Description

We should not show the category or category group if they are empty.

### Are you willing to participate in the development of this feature and create a pull request?

Yes

### PR DESCRIPTION
**Pull request by taroj1205**
Closes #385

## Description

This makes the category without example not show on the page.

## Is this a breaking change (Yes/No):

No

## Additional Information

Please comment if you have suggestion on better way to implement this.

### CODE REVIEW

File: utils/next.ts [hunk 1]
```diff
@@ -28,7 +28,15 @@ export const getServerSideCommonProps = async ({
...
-  const componentTree = await getComponentCategoryGroup()(locale as Locale)
+  const componentTree = (
+    await getComponentCategoryGroup()(locale as Locale)
+  ).filter(({ items }) => items)
+
+  for (const component of componentTree) {
+    if (component.items) {
+      component.items = component.items.filter(({ items }) => items)
+    }
+  }
```
Comment (illionillion) [inline utils/next.ts:39]: How about using the `map` function after `filter`?

```suggestion
const componentTree = (
    await getComponentCategoryGroup()(locale as Locale)
  )
.filter(({ items }) => items)
    .map(component => ({
      ...component,
      items: component.items?.filter(({ items }) => items),
    }))
```

File: utils/next.ts [hunk 2]
```diff
@@ -54,6 +62,12 @@ export const getStaticComponentProps =
       `/${[categoryGroupName, ...paths].join("/")}`,
     )
 
+    for (const component of componentTree) {
```
Comment (illionillion) [inline utils/next.ts:65]: How about using `forEach` rather than `for of`?

Comment (taroj1205) [reply utils/next.ts:65]: @illionillion `for of` loop is considered more performant than `forEach`. What is the reason for using `forEach` over `for of` loop?

Comment (illionillion) [reply utils/next.ts:64]: `forEach` is easier to read than `for of`.
But if performance is better, it's better to unify everything with `for of`.

**Review by illionillion — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**